test/hexagon/include/halide-hexagon-setup.h

```diff
@@ -72,3 +75,27 @@ Expr sat_8(Expr e) {
 Expr usat_8(Expr e) {
   return clamp(e, 0, 255);
 }
+
+Expr saturating_subtract(Expr a, Expr b) {
+  Type wider = Int(a.type().bits() * 2);
+  /* FIXME: There is a problem here that I do not already know how to address.
+    There is no way to saturate and pack shorts (signed or unsigned) into signed
+    bytes. Assuming 'a' and 'b' were, say, i8x64 types, we'd be doing exactly
+    this, i.e. saturating and packing the widened result of the subtract
+    (i16x64) into signed bytes. This isn't supported. Should we be warning here?
```

Comment:
To double check my understanding of the issue being raised here, this is a problem because the HVX instruction set doesn't support this particular combination of saturating arithmetic, correct?

If so, that's fine: the expected behavior should be that the code still works, and that it simply doesn't get optimized to a dedicated instruction if an instruction doesn't exist.